proto/table_marshal.go

```diff
@@ -167,6 +169,17 @@ func (u *marshalInfo) size(ptr pointer) int {
 		u.computeMarshalInfo()
 	}
 
+	// Uses the message's Size method if available
+	if u.hassizer {
```

Comment:
I think, like you suggested, the best we can do is move these checks into the `if u.hasmarshaler`, because then it is quicker to call size than marshaling, but otherwise let's have cachesize run its course.

Reply:
If we move these checks in u.hasmarshaler we wont be calling any other struct field's size methods and thus not init + caching their sizes. I think the best we could do then is just use the default sizing even if it is a sizer, and then only use the Size method if the message is a marshaler as well.